squirrel.frontier/src/test/java/org/dice_research/squirrel/frontier/impl/FrontierImplTest.java

```diff
@@ -126,50 +128,78 @@ public void simlpeRecrawling() throws Exception {
         CrawleableUri uri_2 = cuf.create(new URI("http://dbpedia.org/resource/normalUri"),
-                InetAddress.getByName("127.0.0.1"), UriType.DEREFERENCEABLE);
+            InetAddress.getByName("127.0.0.1"), UriType.DEREFERENCEABLE);
         uris.add(uri_1);
         uris.add(uri_2);
-
         frontier.addNewUris(uris);
-
         List<CrawleableUri> nextUris = frontier.getNextUris();
         for (CrawleableUri uri : nextUris) {
             Assert.assertTrue(uris.contains(uri));
         }
         for (CrawleableUri uri : uris) {
             Assert.assertTrue(nextUris.contains(uri));
         }
-
         // Set the first URI as recrawlable
         for (CrawleableUri uri : nextUris) {
-            if(uri.getUri().equals(uri_1.getUri())) {
+            if (uri.getUri().equals(uri_1.getUri())) {
                 uri.addData(Constants.URI_PREFERRED_RECRAWL_ON, System.currentTimeMillis() - 1);
             }
         }
-
         frontier.crawlingDone(uris);
-        
         uris.add(uri_1);
         uris.add(uri_2);
-
         nextUris = frontier.getNextUris();
         Assert.assertNotNull(nextUris);
         assertTrue("uri_1 has been expected but couldn't be found", nextUris.contains(uri_1));
         Assert.assertEquals(1, nextUris.size());
         assertFalse("uri_2 has been found but was not expected", nextUris.contains(uri_2));
     }
 
+    @Test
+    public void RecrawlingTest() throws Exception {
+        Dataset dataset = DatasetFactory.create();
+        dataset.setDefaultModel(ModelFactory.createDefaultModel());
+        QueryExecutionFactory queryExecFactory = new QueryExecutionFactoryDataset(dataset);
+        UpdateExecutionFactory updateExecFactory = new UpdateExecutionFactoryDataset(dataset);
+
+        CrawleableUri uri = new CrawleableUri(new URI("http://example.org/dataset"));
+        uri.addData(Constants.UUID_KEY, "123");
+        CrawlingActivity activity = new CrawlingActivity(uri, "http://example.org/testRecrawling");
+        uri.addData(Constants.URI_CRAWLING_ACTIVITY, activity);
+      /*  List<CrawleableUri> uris = new ArrayList<>();
+        CrawleableUri uri_1 = cuf.create(new URI("http://dbpedia.org/resource/uriThatShouldBeRecrawled"));
+        uri_1.addData("endedAtTime", "2019-07-06T17:04:02.864Z");
+        CrawleableUri uri_2 = cuf.create(new URI("http://dbpedia.org/resource/normalUri"));
+        uri_2.addData("endedAtTime", "2019-07-06T19:38:02.864Z");
+        uris.add(uri_1);Build);
+        }
+
+        // Set the first URI as recrawlable
+        for (CrawleableUri uri : nextUris) {
+            if (uri.getUri().equals(uri_1.getUri())) {
+                uri.addData(Constants.URI_PREFERRED_RECRAWL_ON, System.currentTimeMillis() - 1);
+            }
+        }
+        Assert.assertNotNull(nextUris);
+        assertTrue("uri_1 has been expected but couldn't be found", nextUris.contains(uri_1));
+        Assert.assertEquals(2, nextUris.size());
+
+        */
```

Comment:
Why is the test commented out? Please make sure that you have a working test :wink: